### PR TITLE
Add network queries package

### PR DESCRIPTION
**Pull request by lamcc21**
Issue #38 
Added queries for #36 but not codegen

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Changes

- add network-queries package, migrate .ts query files to .gql
- update codegen configs and remove query files from react-hooks and network clients
- add workflows and update test file

## TODO

- [x] network-queries need to be published and then `@subql/network-queries` dependency can be added to react-hooks and network-clients packages
- [x] add support for graphql language server
- [x] remove unnecessary dependencies

### CODE REVIEW

File: packages/react-hooks/codegen.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ generates:
   src/:
     preset: near-operation-file
     presetConfig:
-      folder: ../__hooks__
+      folder: ../../../../../src/__hooks__
       extensions: .generated.ts
```
Comment (lamcc21) [inline packages/react-hooks/codegen.yaml:26]: I didn't want to do this but no alternative.

**Comment by HZHAIX** [comment]
@Jianzhuo you can continue this task

**Comment by HZHAIX** [comment]
Need to update this PR as `network-queries`  is on npm now

File: test/queryClient.test.ts [hunk 2]
```diff
@@ -1,36 +1,36 @@
 // Copyright 2020-2022 SubQuery Pte Ltd authors & contributors
 // SPDX-License-Identifier: Apache-2.0
 
-import { GraphqlQueryClient } from '../packages/network-clients';
-import { NETWORK_CONFIGS } from '../packages/network-clients/src';
+import { GetAllDelegations, GraphqlQueryClient } from '@subql/network-clients/src';
```
Comment (HZHAIX) [inline test/queryClient.test.ts:4]: Can we get these from `@subql/network-query`

File: test/queryClient.test.ts [hunk 3]
```diff
@@ -1,36 +1,36 @@
...
+import { NETWORK_CONFIGS } from '@subql/network-clients/src';
```
Comment (HZHAIX) [inline test/queryClient.test.ts:5]: change `@subql/network-clients/src` to `@subql/network-clients`

File: test/queryClient.test.ts [hunk 4]
```diff
@@ -1,36 +1,36 @@
...
-} from "@subql/network-clients/dist/graphql/offers";
-import { GET_DEPLOYMENT_PLANS, GET_PLAN_TEMPLATES, GET_PLANS } from "@subql/network-clients/dist/graphql/plans";
-import { GET_PROJECT, GET_PROJECT_DEPLOYMENTS, GET_PROJECTS } from "@subql/network-clients/dist/graphql/project";
-import { GET_INDEXER_REWARDS, GET_REWARDS, GET_WITHDRAWLS } from "@subql/network-clients/dist/graphql/staking";
+  GetAllOpenOffers,
+  GetOwnExpiredOffers,
+  GetOwnOpenOffers, 
+  GetSpecificOpenOffers
+} from "@subql/network-clients/src";
+import { GetDeploymentPlans, GetPlanTemplates, GetPlans } from "@subql/network-clients/src";
```
Comment (HZHAIX) [inline test/queryClient.test.ts:31]: Same with above questions

File: test/queryClient.test.ts (hunk 2 above)
Comment (lamcc21) [reply test/queryClient.test.ts]: Set alias in monorepo tsconfig to solve this.

File: packages/network-clients/codegen.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
+// Copyright 2020-2022 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { CodegenConfig } from '@graphql-codegen/cli'
```
Comment (HZHAIX) [inline packages/network-clients/codegen.ts:4]: Can we move the types and type generation script to `network-query`, and export the types.
These types are generated from `*.gql`, could be put them in the same package, in that case the query and types can maintain together to make the sure the interface is consistent.

**Comment by HZHAIX** [comment]
@lamcc21 awesome work for these changes, here are some improvement  points basing on our discussion:

- [ ] For `network-clients`: remove the `codegen.ts` script and import the types and queries` from `netowrk-query` package
- [ ] For `network-query`:  include `codegen.ts` and generate the types and js query functions, export them for public use.
- [ ] For `react-hook`: remove the type generation and change the `baseTypesPath` to `@subql/network-query`

**Review by HZHAIX — APPROVED** [review]

**Pull request merged**